src/dscanner/analysis/pointless_arithmetic_check.d

```diff
@@ -0,0 +1,93 @@
+//          Copyright Brian Schott (Hackerpilot) 2016.
+// Distributed under the Boost Software License, Version 1.0.
+//    (See accompanying file LICENSE_1_0.txt or copy at
+//          http://www.boost.org/LICENSE_1_0.txt)
+module dscanner.analysis.pointless_arithmetic_check;
+
+import dparse.ast;
+import dparse.lexer;
+import dscanner.analysis.base;
+import std.stdio;
+import std.typetuple;
+
+/**
+ * Checks for Pointless Arithmetic.
+ */
+final class PointlessArithmeticCheck : BaseAnalyzer
+{
+	alias visit = BaseAnalyzer.visit;
+
+	///
+	this(string fileName, bool skipTest = false)
+	{
+		super(fileName, null, skipTest);
+	}
+
+	override void visit(const MulExpression mulExp)
+	{
+		UnaryExpression left = cast(UnaryExpression) mulExp.left;
+		UnaryExpression right = cast(UnaryExpression) mulExp.right;
+
+		if (mulExp.operator == tok!"/" &&
```

Comment:
Please protect with `if (left && right){ ... }`

Reply:
oh, I missed. Thank you.